src/zenml/utils/dashboard_utils.py

```diff
@@ -12,69 +12,52 @@
 #  or implied. See the License for the specific language governing
 #  permissions and limitations under the License.
 """Utility class to help with interacting with the dashboard."""
-from functools import partial
 from typing import Optional
-from uuid import UUID
 
 from zenml.client import Client
 from zenml.constants import ENV_AUTO_OPEN_DASHBOARD, handle_bool_env_var
 from zenml.enums import EnvironmentType, StoreType
 from zenml.environment import get_environment
 from zenml.logger import get_logger
-from zenml.utils.pagination_utils import depaginate
+from zenml.models.pipeline_run_models import PipelineRunResponseModel
 
 logger = get_logger(__name__)
 
 
-def get_run_url(
-    run_name: str, pipeline_id: Optional[UUID] = None
-) -> Optional[str]:
+def get_run_url(run: PipelineRunResponseModel) -> Optional[str]:
     """Computes a dashboard url to directly view the run.
 
     Args:
-        run_name: Name of the pipeline run.
-        pipeline_id: Optional pipeline_id, to be sent when available.
+        run: Pipeline run to be viewed.
 
     Returns:
         A direct url link to the pipeline run details page. If run does not exist,
         returns None.
     """
-    # Connected to ZenML Server
     client = Client()
 
     if client.zen_store.type != StoreType.REST:
         return ""
 
     url = client.zen_store.url
-    runs = depaginate(partial(client.list_runs, name=run_name))
+    pipeline_id = str(run.pipeline.id)
+    run_id = str(run.id)
 
-    if pipeline_id:
-        url += f"/workspaces/{client.active_workspace.name}/pipelines/{str(pipeline_id)}/runs"
-    elif runs:
-        url += "/runs"
-    else:
-        url += "/pipelines/all-runs"
-
-    if runs:
-        url += f"/{runs[0].id}/dag"
+    url += f"/workspaces/{client.active_workspace.name}/pipelines/{pipeline_id}/runs/{run_id}/dag"
```

Comment:
Someone will generally want to have the view that ties their run back to the pipeline, I figured, so that's why I chose this URL vs the disembodied 'one run of many runs' option.

Reply:
My assumption is that if we're passing a 'run' object into the function, we'll always have a run ID and a pipeline ID available to us.

Reply:
We thought about this when implementing the pipeline versioning and back then the consensus was to keep the unlisted runs. I guess it doesn't really hurt us to keep the feature and maybe some users don't want to "clutter" their pipeline overview page with random local experiments that never make it any further?

Don't have a strong opinion on this though, if we find a good reason for it (e.g. it would make the code much easier) I'm also open to deprecating and removing the feature at some point.